openupgradelib/openupgrade.py

```diff
@@ -2367,3 +2368,88 @@ def set_xml_ids_noupdate_value(env, module, xml_ids, value):
         SET noupdate = %s
         WHERE module = %s AND name in %s
     """, (value, module, tuple(xml_ids),))
+
+def move_to_company_dependent(
+        env,
+        model_name,
+        model_table_name,
+        origin_field_name,
+        destination_field_name,
+    ):
+        """ For each row in a given table, the value of a given field is
+        set in another 'company dependant' field of the same table.
+        Useful in cases when from one version to another one, some field in a
+        model becomes a 'company dependent' field.
+
+        :param model_name: Name of the model.
+        :param model_table_name: Name of the table.
+        :param origin_field_name: Name of the field from which the values
+                                  will be obtained.
+        :param company_dependent_field_name: Name of the 'company dependent'
+                                             field where the values obtained
+                                             from origin_field_name will be
+                                             set.
+        """
+        logger.debug(
+            "Moving data to a company_dependent field:\n"
+            "   Model: {}\n"
+            "   Origin field name: {}\n"
+            "   Destination field name: {}".format(
+                model_name, origin_field_name, destination_field_name
+            )
+        )
+        cr = env.cr
+        mapping_type2field = {
+            'char': 'value_text',
+            'float': 'value_float',
+            'boolean': 'value_integer',
+            'integer': 'value_integer',
+            'text': 'value_text',
+            'binary': 'value_binary',
+            'many2one': 'value_reference',
+            'date': 'value_datetime',
+            'datetime': 'value_datetime',
+            'selection': 'value_text',
+        }
+        # Determine field id and the model name of the relation
+        # in case of many2one.
+        cr.execute("""
```

Comment:
Instead of this (and if not provided as argument), do `env[origin_field_name]._fields[model_name].type`